chain/x/zoracle/internal/types/msgs.go

```diff
@@ -61,6 +61,72 @@ func (msg MsgRequest) GetSignBytes() []byte {
 	return sdk.MustSortJSON(bz)
 }
 
+// MsgRequestData is a new version of request message.
+type MsgRequestData struct {
+	OracleScriptID           int64          `json:"oracleScriptID"`
+	Calldata                 []byte         `json:"calldata"`
+	RequestedValidatorCount  int64          `json:"requestedValidatorCount"`
+	SufficientValidatorCount int64          `json:"sufficientValidatorCount"`
+	Expiration               int64          `json:"expiration"`
+	Sender                   sdk.AccAddress `json:"sender"`
+}
+
+// NewMsgRequestData is a constructor function for MsgRequestData
+func NewMsgRequestData(
+	oracleScriptID int64,
+	calldata []byte,
+	requestedValidatorCount int64,
+	sufficientValidatorCount int64,
+	expiration int64,
+	sender sdk.AccAddress,
+) MsgRequestData {
+	return MsgRequestData{
+		OracleScriptID:           oracleScriptID,
+		Calldata:                 calldata,
+		RequestedValidatorCount:  requestedValidatorCount,
+		SufficientValidatorCount: sufficientValidatorCount,
+		Expiration:               expiration,
+		Sender:                   sender,
+	}
+}
+
+// Route should return the name of the module
+func (msg MsgRequestData) Route() string { return RouterKey }
+
+// Type should return the action
+func (msg MsgRequestData) Type() string { return "request" }
+
+// ValidateBasic runs stateless checks on the message
+func (msg MsgRequestData) ValidateBasic() sdk.Error {
+	if msg.Sender.Empty() {
```

Comment:
I will have a ticket to revisit all error messages later. For now it's low priority.

Reply:
Sir yes sir